File: NDde.2.01.0563.src/NDde/Samples/cs/ClientWin/MainForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using NDde.Client;

namespace ClientWin
{
    public partial class MainForm : Form
    {
        private DdeClient client;

        public MainForm()
        {
            InitializeComponent();

            client = new DdeClient("myapp", "myservice", this);
            client.Advise += client_Advise;
            client.Disconnected += client_Disconnected;
        }

        private void MainForm_Load(object sender, EventArgs e)
        {
            try
            {
                // Connect to the server.  It must be running or an exception will be thrown.
                client.Connect();

                // Advise Loop
                client.StartAdvise("myitem", 1, true, 60000);
            }
            catch (Exception ex)
            {
                displayTextBox.Text = "MainForm_Load: " + ex.Message;
            }
        }

        private void MainForm_Resize(object sender, EventArgs e)
        {
            displayTextBox.Left = this.DisplayRectangle.Left;
            displayTextBox.Width = this.DisplayRectangle.Width;
            displayTextBox.Top = this.DisplayRectangle.Top;
            displayTextBox.Height = this.DisplayRectangle.Height;
        }

        private void client_Advise(object sender, DdeAdviseEventArgs args)
        {
            displayTextBox.Text = "OnAdvise: " + args.Text;
        }

        private void client_Disconnected(object sender, DdeDisconnectedEventArgs args)
        {
            displayTextBox.Text = 
                "OnDisconnected: " +
                "IsServerInitiated=" + args.IsServerInitiated.ToString() + " " +
                "IsDisposed=" + args.IsDisposed.ToString();
        }

    } // class

} // namespace

File: NDde.2.01.0563.src/NDde/Samples/vb/ClientWin/MainForm.vb
Imports System.Text
Imports NDde.Client

Public Class MainForm
    Private WithEvents client As New DdeClient("myapp", "mytopic", Me)

    Private Sub MainForm_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load
        Try
            ' Connect to the server.  It must be running or an exception will be thrown.
            client.Connect()

            ' Advise Loop
            client.StartAdvise("myitem", 1, True, 60000)
        Catch ex As Exception
            displayTextBox.Text = ex.Message
        End Try
    End Sub

    Private Sub MainForm_Resize(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Resize
        displayTextBox.Left = Me.DisplayRectangle.Left
        displayTextBox.Width = Me.DisplayRectangle.Width
        displayTextBox.Top = Me.DisplayRectangle.Top
        displayTextBox.Height = Me.DisplayRectangle.Height
    End Sub

    Private Sub client_Advise(ByVal sender As Object, ByVal e As NDde.Client.DdeAdviseEventArgs) Handles client.Advise
        displayTextBox.Text = "OnAdvise: " + e.Text
    End Sub

    Private Sub client_Disconnected(ByVal sender As Object, ByVal e As NDde.Client.DdeDisconnectedEventArgs) Handles client.Disconnected
        displayTextBox.Text = _
         "OnDisconnected: " + _
         "IsServerInitiated=" + e.IsServerInitiated.ToString() + " " + _
         "IsDisposed=" + e.IsDisposed.ToString()
    End Sub
End Class
